src/random-seed/random-seed.c

```
@@ -306,6 +308,69 @@ static int save_seed_file(
         return 0;
 }
 
+static int refresh_boot_seed(void) {
+        uint8_t buffer[RANDOM_EFI_SEED_SIZE];
+        struct sha256_ctx hash_state;
+        _cleanup_free_ char *esp_path = NULL, *seed_path = NULL;
+        _cleanup_free_ void *seed_file_bytes = NULL;
+        _cleanup_close_ int seed_fd = -1;
+        size_t len;
+        ssize_t r;
+
+        assert_cc(RANDOM_EFI_SEED_SIZE == SHA256_DIGEST_SIZE);
+
+        /* We're doing this opportunistically, so if the seeding dance before didn't manage to initialize the
+         * RNG, there's no point in doing it here. Secondly, getrandom(GRND_NONBLOCK) has been around longer
+         * than EFI seeding anyway, so there's no point in having non-getrandom() fallbacks here. So if this
+         * fails, just return early to cut our losses. */
+        r = getrandom(buffer, sizeof(buffer), GRND_NONBLOCK);
+        if (r != sizeof(buffer))
+                return r < 0 ? -errno : -EIO;
```

maybe:

```c
if (r == -EAGAIN) {
        log_debug_errno(r, "Random pool not initialized yet, skipping ESP random seed update.");
        return 0;
}       
if (r < 0)
        return log_error_errno(r, "Failed to acquire random seed: %m");
assert(r == sizeof(buffer));
```

(i.e. debug log about expected error, and fail loudly about unexpected error, and abort on impossible error)

(I mean, short returns are not possible with getrandom() on short buffers like this, right?)

---

Will do. Yes, short returns not possible for 32 bytes.